internal/virtual/file.go

```diff
@@ -0,0 +1,67 @@
+package virtual
+
+import (
```

Comment:
This is a copy from both conjure and fscache. I'd like to consolidate those implementations into virtual.